gnmi_server/gnoi.go

```diff
@@ -20,6 +20,40 @@ import (
 	jwt "github.com/dgrijalva/jwt-go"
 )
 
+func KillOrRestartProcess(restart bool, serviceName string) error {
+	sc, err := ssc.NewDbusClient()
+	if err != nil {
+		return err
+	}
+	if restart {
+		log.V(2).Infof("Restarting service %s...", serviceName)
+		err = sc.RestartService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to restart service %s: %v", serviceName, err)
+		}
+	} else {
+		log.V(2).Infof("Stopping service %s...", serviceName)
+		err = sc.StopService(serviceName)
+		if err != nil {
+			log.V(2).Infof("Failed to stop service %s: %v", serviceName, err)
+		}
+	}
+	return err
+}
+
+func (srv *Server) KillProcess(ctx context.Context, req *gnoi_system_pb.KillProcessRequest) (*gnoi_system_pb.KillProcessResponse, error) {
+	serviceName := req.GetName()
+	restart := req.GetRestart()
```

Comment:
>GetRestart [](http://example.com/codeflow?start=16&length=10)

I see other useful fields like `signal`. Could you check if we need to parse them all? We can implement only limited fields, and return NotImplemented for other combination.

Reply:
GNOI protocol defines fields that may be helpful, but there is no requirement to parse them all. I've added additional processing to output Unimplemented if one of the extraneous fields is passed (`signal` or `pid`) by the client.

Reply:
let's treat no signal same as default signal (SIGTERM)

Reply:
Discussed offline, the agreed-upon implementation is to require the client to specify signal, as the protocol defines an unspecified signal as the "invalid default" ref: https://github.com/openconfig/gnoi/blob/349b3dcd04ecf78d740a25f17f983a8c8cd56e73/system/system.pb.go#L243

Because our KillProcess implementation uses generic `systemctl stop` and `restart` commands (which use SIGNAL_TERM signal by default), we require the client to specify the signal option 1: SIGNAL_TERM